### ISSUE TITLE
Demo runtime errors

### ISSUE THREAD

**Issue by iursevla** [PaloAltoNetworks/docusaurus-openapi-docs#980]
## Describe the bug

I was trying to use this project into another open source project that uses docusaurus. I wasn't able to do it since it would always throw an exception when I tried to access a OpenAPI endpoint. To make sure I wasn't missing something I tried to clone this project and run the `demo` page, but it seems to not work locally. 

The front end starts without any error whatsoever. But when I try to access any OpenAPI endpoint it fails. I've tried with Chrome, Safari and Brave and it fails in all of them.

## Steps to reproduce

1. Clone project
2. cd into demo
3. run `yarn install`
4. run `yarn start`

## Screenshots

<img width="1369" alt="image" src="https://github.com/user-attachments/assets/6809ea11-ef5b-4721-a079-e8dbe5c11cf0">

<img width="1409" alt="image" src="https://github.com/user-attachments/assets/d2b6b562-b38b-4160-86af-d44c761385ec">

<img width="1180" alt="image" src="https://github.com/user-attachments/assets/58ea6161-d420-4b34-b4d1-5c8c16462470">

<details>
<summary>Error in Chrome</summary>

<img width="1474" alt="image" src="https://github.com/user-attachments/assets/af069415-9baa-45b1-973a-c6710b232c17">

```text
2index.js:24 Uncaught TypeError: Cannot read properties of undefined (reading 'map')
    at eval (index.js:24:1806)
    at Array.map (<anonymous>)
    at CodeSnippets (index.js:24:621)
    at renderWithHooks (react-dom.development.js:15487:18)
    at mountIndeterminateComponent (react-dom.development.js:20099:13)
    at beginWork (react-dom.development.js:21622:16)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4165:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4214:16)
    at invokeGuardedCallback (react-dom.development.js:4278:31)
    at beginWork$1 (react-dom.development.js:27486:7)
```

```text
react-dom.development.js:18705 The above error occurred in the <CodeSnippets> component:

    at CodeSnippets (webpack-internal:///./node_modules/docusaurus-theme-openapi-docs/lib/theme/ApiExplorer/CodeSnippets/index.js:11:2980)
    at ApiExplorer (webpack-internal:///./node_modules/docusaurus-theme-openapi-docs/lib/theme/ApiExplorer/index.js:6:899)
    at BrowserOnly (webpack-internal:///./node_modules/@docusaurus/core/lib/client/exports/BrowserOnly.js:16:23)
    at div
    at div
    at Provider (webpack-internal:///./node_modules/react-redux/es/components/Provider.js:19:20)
    at MDXProvider (webpack-internal:///./node_modules/@mdx-js/react/lib/index.js:76:18)
    at MDXContent (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/MDXContent/index.js:15:25)
    at div
    at DocItemContent (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/DocItem/Content/index.js:27:311)
    at article
    at div
    at div
    at div
    at DocItemLayout (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/DocItem/Layout/index.js:28:755)
    at HtmlClassNameProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/metadataUtils.js:29:46)
    at DocProvider (webpack-internal:///./node_modules/@docusaurus/plugin-content-docs/lib/client/doc.js:24:26)
    at ApiItem (webpack-internal:///./node_modules/docusaurus-theme-openapi-docs/lib/theme/ApiItem/index.js:8:69)
    at RouteContextProvider (webpack-internal:///./node_modules/@docusaurus/core/lib/client/routeContext.js:16:60)
    at LoadableComponent (webpack-internal:///./node_modules/react-loadable/lib/index.js:138:32)
    at Route (webpack-internal:///./node_modules/react-router/esm/react-router.js:648:29)
    at Switch (webpack-internal:///./node_modules/react-router/esm/react-router.js:850:29)
    at div
    at main
    at DocRootLayoutMain (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/DocRoot/Layout/Main/index.js:16:32)
    at div
    at div
    at DocRootLayout (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/DocRoot/Layout/index.js:18:28)
    at DocsSidebarProvider (webpack-internal:///./node_modules/@docusaurus/plugin-content-docs/lib/client/docsSidebar.js:19:34)
    at HtmlClassNameProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/metadataUtils.js:29:46)
    at DocRoot (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/DocRoot/index.js:20:148)
    at RouteContextProvider (webpack-internal:///./node_modules/@docusaurus/core/lib/client/routeContext.js:16:60)
    at LoadableComponent (webpack-internal:///./node_modules/react-loadable/lib/index.js:138:32)
    at Route (webpack-internal:///./node_modules/react-router/esm/react-router.js:648:29)
    at Switch (webpack-internal:///./node_modules/react-router/esm/react-router.js:850:29)
    at DocsVersionProvider (webpack-internal:///./node_modules/@docusaurus/plugin-content-docs/lib/client/docsVersion.js:17:34)
    at HtmlClassNameProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/metadataUtils.js:29:46)
    at DocVersionRootContent (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/DocVersionRoot/index.js:18:829)
    at DocVersionRoot
    at RouteContextProvider (webpack-internal:///./node_modules/@docusaurus/core/lib/client/routeContext.js:16:60)
    at LoadableComponent (webpack-internal:///./node_modules/react-loadable/lib/index.js:138:32)
    at Route (webpack-internal:///./node_modules/react-router/esm/react-router.js:648:29)
    at Switch (webpack-internal:///./node_modules/react-router/esm/react-router.js:850:29)
    at ErrorBoundary (webpack-internal:///./node_modules/@docusaurus/core/lib/client/exports/ErrorBoundary.js:16:269)
    at div
    at NavbarSecondaryMenuDisplayProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/navbarSecondaryMenu/display.js:21:318)
    at NavbarMobileSidebarProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/navbarMobileSidebar.js:23:414)
    at NavbarSecondaryMenuContentProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/navbarSecondaryMenu/content.js:16:157)
    at NavbarProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/navbarUtils.js:22:29)
    at HtmlClassNameProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/metadataUtils.js:29:46)
    at PluginHtmlClassNameProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/metadataUtils.js:32:42)
    at DocsPreferredVersionContextProviderUnsafe (webpack-internal:///./node_modules/@docusaurus/plugin-content-docs/lib/client/docsPreferredVersion.js:32:382)
    at DocsPreferredVersionContextProvider (webpack-internal:///./node_modules/@docusaurus/plugin-content-docs/lib/client/docsPreferredVersion.js:35:50)
    at ScrollControllerProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/scrollUtils.js:21:489)
    at AnnouncementBarProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/announcementBar.js:26:450)
    at ColorModeProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/colorMode.js:28:1246)
    at eval (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/reactUtils.js:59:9)
    at LayoutProvider (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/Layout/Provider/index.js:20:651)
    at Layout (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/Layout/index.js:26:33)
    at HtmlClassNameProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/metadataUtils.js:29:46)
    at DocsRoot (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib/theme/DocsRoot/index.js:18:544)
    at RouteContextProvider (webpack-internal:///./node_modules/@docusaurus/core/lib/client/routeContext.js:16:60)
    at LoadableComponent (webpack-internal:///./node_modules/react-loadable/lib/index.js:138:32)
    at Route (webpack-internal:///./node_modules/react-router/esm/react-router.js:648:29)
    at Switch (webpack-internal:///./node_modules/react-router/esm/react-router.js:850:29)
    at Route (webpack-internal:///./node_modules/react-router/esm/react-router.js:648:29)
    at ClientLifecyclesDispatcher (webpack-internal:///./node_modules/@docusaurus/core/lib/client/ClientLifecyclesDispatcher.js:15:261)
    at PendingNavigation (webpack-internal:///./node_modules/@docusaurus/core/lib/client/PendingNavigation.js:18:150)
    at AppNavigation (webpack-internal:///./node_modules/@docusaurus/core/lib/client/App.js:29:252)
    at Root (webpack-internal:///./node_modules/@docusaurus/core/lib/client/theme-fallback/Root/index.js:20:16)
    at BrowserContextProvider (webpack-internal:///./node_modules/@docusaurus/core/lib/client/browserContext.js:21:127)
    at DocusaurusContextProvider (webpack-internal:///./node_modules/@docusaurus/core/lib/client/docusaurusContext.js:21:496)
    at ErrorBoundary (webpack-internal:///./node_modules/@docusaurus/core/lib/client/exports/ErrorBoundary.js:16:269)
    at App
    at Router (webpack-internal:///./node_modules/react-router/esm/react-router.js:267:30)
    at BrowserRouter (webpack-internal:///./node_modules/react-router-dom/esm/react-router-dom.js:58:35)
    at Router (webpack-internal:///./node_modules/@docusaurus/core/lib/client/clientEntry.js:19:21)
    at r (webpack-internal:///./node_modules/react-helmet-async/lib/index.module.js:18:7350)

React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary.
```

</details>

## Your Environment

- Version used:
- MacOS 14.5
- Chrome Version 125.0.6422.1
- Safari Version 17.5 (19618.2.12.11.6)
- Brave Version 1.69.168 Chromium: 128.0.6613.138 (Official Build) (arm64)

**Comment by iursevla** [comment]
@sserrata can you help on this? Are you guys able to run the demo that is currently on the master branch? I can't make it work properly (see errors above).

**Comment by sserrata** [comment]
Hi @iursevla, the demo is currently live on the https://docusaurus-openapi.tryingpan.dev/ site. Are you looking to contribute or simply use the plugin in a Docusaurus project/site? If you're looking to simply use the plugin, please follow the installation steps in the docs. Thanks!

**Comment by iursevla** [comment]
@sserrata the problem is that I can't run the demo locally. Basically when I download the demo directory and then

`yarn install` + `yarn start` it starts throwing errors at runtime. What could I be doing wrong to not be able to properly use the demo?

<img width="2239" alt="image" src="https://github.com/user-attachments/assets/87f2ded8-9be3-48be-beae-706e8b6959e4">

**Comment by iursevla** [comment]
@sserrata seems like it's fixed on 4.1.0. I was not able to use 4.0.1 properly. I'm gonna close this issue.
